In [ ]:
#import necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats

In [ ]:
from google.colab import files

In [ ]:
# this code allows us to upload files from our local device
uploaded = files.upload()

Saving medication_data.csv to medication_data.csv


In [ ]:
import io

In [ ]:
# read the uploaded csv file into a pandas dataframe
medication_df = pd.read_csv(io.BytesIO(uploaded['medication_data.csv']))

In [ ]:
# view first five rows of the dataframe
medication_df.head()

Group  Score
0   0mg      9
1   0mg      8
2   0mg      7
3   0mg      8
4   0mg      8

In [ ]:
# group the data by the column "Group"
grouped_df = medication_df.groupby("Group")

### ANOVA

In [ ]:
k = len(grouped_df.nunique()) #no of data groups
n = grouped_df.size()[0] #population of data per group
N = k * n #total population of data

In [ ]:
#Degree of Freedom
DF_total = N - 1
DF_between = k - 1
DF_within = N - k

In [ ]:
# Calculate Group Mean (GM)
GM = grouped_df.Score.mean()

In [ ]:
# Calculate Population Mean (PM)
PM = medication_df.Score.mean()

In [ ]:
# Calculate Sum of Squares (SS)
SS_between = (((GM - PM) ** 2) * n).sum()
SS_within = sum(medication_df['Score'].values ** 2) - sum(grouped_df.sum()['Score']**2)/n
SS_total = ((medication_df['Score'].values - PM) ** 2).sum()

In [ ]:
# Calculate Mean Sum of Squares
MS_between = SS_between / DF_between
MS_within = SS_within / DF_within

In [ ]:
# Calculate F-Value
F_value = MS_between/MS_within

In [ ]:
# Calculate p_value
p_value = stats.f.sf(F_value, DF_between, DF_within)

In [ ]:
print(F_value,p_value)

86.3333333333329 5.956341358737707e-10


## Interpretation:
The calculated value of F is 86.3333 which is significantly greater than the critical value of F i.e. 3.5546.
As F_value > F_critical_value, null hypothesis is rejected and alternative hypothesis is accepted.

## Conclusion:
The three different groups of medications used in the experiment do not have the same effect.